test/auth_test.rb

```diff
@@ -209,62 +209,71 @@ def app
       assert_equal 7200, tokens["expires_in"]
     end
 
-    it "accepts an authorization code" do
-      i = 0
-      stub_heroku_api do
-        post("/oauth/tokens") {
-          # only check parameters on the 2nd request, as ID does one OAuth
-          # dance to procure its own token
-          if (i += 1) > 1
```

Comment:
I tested this, and the assumption in the comment isn't true. lines 219-221 are never executed with this check in place.